### PR TITLE
`EventProp` is a `Functor`

### PR DESCRIPTION
**Pull request by armanbilge**
Closes https://github.com/armanbilge/calico/issues/167.

I think this is a really interesting idea. I'm a bit worried if it makes `EventProp` a more confusing API, but I am not sure how user-facing it is anyway.

### CODE REVIEW

**Comment by armanbilge** [comment]
Does this mean that `Prop` and `HtmlAttr` should implement `Contramap`? Seems like it.

**Comment by hejfelix** [comment]
Yeah, I would say so, even if I don't have a use case for that yet.

File: calico/src/main/scala/calico/html/Prop.scala [hunk 1]
```diff
@@ -133,15 +136,32 @@ private trait PropModifiers[F[_]](using F: Async[F]):
...
 object EventProp:
-  final class PipeModifier[F[_], E](
+  def apply[F[_], E](key: String): EventProp[F, E, E] =
+    new EventProp(key, identity(_))
+
+  final class PipeModifier[F[_], E] private[calico] (
       private[calico] val key: String,
       private[calico] val sink: Pipe[F, E, Nothing])
 
+  inline given [F[_], E]: (Functor[EventProp[F, E, _]] & FunctorFilter[EventProp[F, E, _]]) =
```
Comment (2chilled) [inline calico/src/main/scala/calico/html/Prop.scala:156]: Intersection type instance, exciting :smile:

**Review by 2chilled — APPROVED** [review]

**Comment by armanbilge** [comment]
> Yeah, I would say so, even if I don't have a use case for that yet.

- https://github.com/armanbilge/calico/pull/169

**Comment by armanbilge** [comment]
Ok, I hope I don't regret this 😁

**Pull request merged**